posts/2021-07-08-sig-bugs.adoc

```diff
@@ -0,0 +1,133 @@
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17235[Improve featureUtility return code when invalid action name is specified]
++
+Users would see an erroneous `return code 0` when an invalid action was specified. This was now changed to `BAD_ARGUMENT (20)`. Because of this change any scripted invocations of featureUtility can now detect this.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17299[Allow multiple version of singleton feature with featureUtility installFeature command]
++
+FeatureUtility installFeature command did not support the installation of multiple versions of the singleton feature using a single command. As a workaround, users had to install each version of the feature one by one. The fix will now allow users to install multiple versions of the feature in a single command, and will also allow users to install a feature bundle addon (libertyCoreBundle, baseBundle, etc) to install all features that apply to their liberty edition. 
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17666[JavaMail tries to use a resource file that only exists in the implementation]
++
+Instead of the `javax.mail.Session` constructor (from `wlp/dev/api/spec/com.ibm.websphere.javaee.mail.1.6_1.0.53.jar`) tried to load the resource `/META-INF/javamail.default.address.map`, that is required. However, this file only existed in the implementation (`wlp/lib/com.ibm.ws.com.sun.mail.javax.mail.1.6_1.6.53.jar`) and it is not able to find it, so it caused a warning to show up:
+
+[source, text]
+----
+[WARNING ] expected resource not found: /META-INF/javamail.default.address.map
+----
+
+This issue was resolved by adding `javamail.default.address.map` to the `app-resource` header.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17658[concurrencyPolicy loses queue slots when managed executor deactivates and erroneously cancels tasks of other executors]
++
+It has been noticed that the identifier that is supplied by the managed executor to `PolicyExecutor.cancel` is ignored by the policy executor and it appears to indiscriminately cancel everything. This would cause a bug where 2 or more managed executors sharing the same `concurrencyPolicy` and used by different applications where only one managed executor deactivates, would erroneously see cancellation of tasks belonging to the other managed executor as well. We have now switched the code from removing directly from the queue and then issuing the cancel to just issuing the cancel.  This is the correct solution because the cancel operation contains the correct logic for decrementing the count against the `maxQueueSize`.
```

Comment:
I would write out the word two here instead of the number

Reply:
I think we can remove the words 'as well'.  The sentence is already really really long and I don't think it adds anything new.

Reply:
Done